lib/logstash/inputs/elasticsearch.rb

```diff
@@ -175,15 +149,48 @@ def run(output_queue)
 
     # since 'scan' doesn't return data on the search call, do an extra scroll
     if @scan
-      resp = run_next(output_queue, r['_scroll_id'])
+      r = run_next(output_queue, r['_scroll_id'])
+      has_hits = r['has_hits']
+    else # not a scan, process the response
+      r['hits']['hits'].each { |hit| push_hit(hit, output_queue) }
+      has_hits = r['hits']['hits'].any?
     end
 
-    while resp[:has_hits] do
-      resp = run_next(output_queue, resp[:scroll_id])
+    while has_hits do
+      r = run_next(output_queue, r['_scroll_id'])
+      has_hits = r['has_hits']
     end
   end # def run
 
   private
+  def run_next(output_queue, scroll_id)
```

Comment:
I know this is 100% my fault :) but maybe we should rename this method to `process_next_scroll` or something like that. what do you think? seeing run_next in these random branchings confuse me as to what I'm actually calling